src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -36,6 +37,9 @@ protected override void HandleWebResourceRequest(ICoreWebView2WebResourceRequest
 				eventArgs.ResourceContext == CoreWebView2WebResourceContextWrapper.Document ||
 				eventArgs.ResourceContext == CoreWebView2WebResourceContextWrapper.Other; // e.g., dev tools requesting page source
 
+			// Get a deferral object so that WebView2 knows there's some async stuff going on. We call Complete() at the end of this method.
+			var deferral = eventArgs.GetDeferral();
```

Comment:
This is the important part, plus the call to `Complete()` at the end of this method.